api/v1alpha1/managedfleetnotificationrecord_types.go

```diff
@@ -171,33 +171,59 @@ func (fnr *ManagedFleetNotificationRecord) CanBeSent(mc, name, clusterID string)
 }
 
 // AddNotificationRecordItem adds a new record item to the notification record slice
-func (fnr *ManagedFleetNotificationRecord) AddNotificationRecordItem(clusterID string, rn *NotificationRecordByName) {
-	ri := NotificationRecordItem{
-		ServiceLogSentCount: 0,
-		HostedClusterID:     clusterID,
-		LastTransitionTime:  nil,
+func (fnr *ManagedFleetNotificationRecord) AddNotificationRecordItem(clusterID string, rn *NotificationRecordByName) (*NotificationRecordItem, error) {
+	for i, nfr := range fnr.Status.NotificationRecordByName {
+		if nfr.NotificationName != rn.NotificationName {
+			continue
+		}
+		for _, nfi := range nfr.NotificationRecordItems {
+			if nfi.HostedClusterID == clusterID {
+				return nil, fmt.Errorf("notification record item for notification %v and hosted cluster %v exists", rn.NotificationName, clusterID)
+			}
+		}
+		ri := NotificationRecordItem{
+			ServiceLogSentCount: 0,
+			HostedClusterID:     clusterID,
+			LastTransitionTime:  nil,
+		}
+		fnr.Status.NotificationRecordByName[i].NotificationRecordItems = append(fnr.Status.NotificationRecordByName[i].NotificationRecordItems, ri)
+		return &ri, nil
 	}
-
-	rn.NotificationRecordItems = append(rn.NotificationRecordItems, ri)
+	return nil, fmt.Errorf("notification %v does not exist", rn.NotificationName)
```

Comment:
Thanks Matt. Just a small note here. The Update and Remove functions are fully covered by tests so far. But for the Add method, this line isn't covered yet along with the `continue` statement on line 177. I think the continue statement is comparatively minor but might help to have additional test case for the `return` statement here. WDYT?

Reply:
Yep, that's a fair point - added a test for it.